specification/integration_context.md

```diff
@@ -0,0 +1,91 @@
+<!-- markdownlint-configure-file 
+{ 
+  "MD013": { 
+    "tables": false 
+  } 
+}
+-->
+
+# Integration Context
+
+**Status**: [Experimental](../README.md#versioning-and-status-of-the-specification)
+
+In order to create a more complete picture of a distributed environment, we
+propose the use of additional headers to pass extra context details between
+Cisco observability components. Specifically, we are concerned with passing
+information between OpenTelemetry-based Splunk distributions and AppDynamics
+agents, in both directions.
+
+# Headers
+
+The following defines several new headers and describes their intended use.
+All headers MUST be treated as optional -- peer services will not always
+generate them. These headers SHOULD be treated as opaque values of type
+string.
```

Comment:
What means opaque here?
Should internal.public API implementing it treat it as an additional class with string value field, liek the Baggage + metadata?

Reply:
This sentence should be indeed clarified. There is too much left for interpretation.

I think the intention is to have a dedicated opaque type for each header.
Thanks to it in future you can add funtionality to use  `cisco-ctx-app-id` as `int` and leave other intact.

From https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/baggage/api.md#set-value

>  `Metadata` [..] This should be an opaque wrapper for a string with no semantic meaning. Left opaque to allow for future functionality.

Reply:
to me opaque here means that agents should not attempt to parse or otherwise derive a meaning for these headers, they are just values as far as agent is concerned

Reply:
> to me opaque here means that agents should not attempt to parse or otherwise derive a meaning for these headers, they are just values as far as agent is concerned

@laurit is exactly right -- the intent is that it's of type string and that it shouldn't need to be parsed. I think the use of "opaque" to mean this is common.

Reply:
> I think the intention is to have a dedicated opaque type for each header.
Thanks to it in future you can add funtionality to use cisco-ctx-app-id as int and leave other headers intact.

No, I don't want any of this spec'd as anything other than a string.

Reply:
>  the intent is that it's of type string and that it shouldn't need to be parsed. I think the use of "opaque" to mean this is common.

I strongly disagree. Opaque is more like encapsulation and information hiding.

I just found https://en.wikipedia.org/wiki/Opaque_data_type to back up my understanding.

Also from Oxford Dictionary:

> opaque - (adj.) not able to be seen through; not transparent.

Reply:
> Then let's spec it as "of string type"

As I wrote it, it literally says "opaque values of type string" but you seem hung up on the use of "opaque". 

Anyway, I like the summary in this stack overflow post:
<img width="669" alt="image" src="https://github.com/user-attachments/assets/fce872ce-a6bc-4a3e-8575-d2054cfcd6c2" />

at least that is the exact same sentiment I was aiming for. It's just some characters, don't parse it, don't extract meaning from those bytes. Just pass it along.

Reply:
I am fine with the current wording. I know realized that it is in the used in the context where we talk about the meaning of the header values and not how we want to design/define types in our distros.

@Kielek, PTAL